zeebe/gateway-grpc/src/test/java/io/camunda/zeebe/gateway/interceptors/impl/AuthenticationInterceptorTest.java

```diff
@@ -426,7 +426,117 @@ public void addsClientIdInOidcToContext() {
   }
 
   @Test
-  public void addsClientIdInPreferenceToUsernameInOidcToContext() {
+  public void addsUsernameInPreferenceToClientIdWhenPreferUsernameClaimIsTrue() {
+    // given
+    final Metadata metadata = createAuthHeader();
+    final CloseStatusCapturingServerCall closeStatusCapturingServerCall =
+        new CloseStatusCapturingServerCall();
+
+    // Create a mock JWT with claims
+    final var jwt = mock(org.springframework.security.oauth2.jwt.Jwt.class);
+    final Map<String, Object> claims = Map.of("username", "test-user", "application_id", "app-id");
+    when(jwt.getClaims()).thenReturn(claims);
+
+    // Configure the JwtDecoder to return our mock JWT
+    final var jwtDecoder = mock(JwtDecoder.class);
+    when(jwtDecoder.decode(anyString())).thenReturn(jwt);
+
+    final var oidcAuthenticationConfiguration = new OidcAuthenticationConfiguration();
+    oidcAuthenticationConfiguration.setUsernameClaim("username");
+    oidcAuthenticationConfiguration.setClientIdClaim("application_id");
+    oidcAuthenticationConfiguration.setPreferUsernameClaim(true);
+
+    // when
+    new AuthenticationInterceptor(new Oidc(jwtDecoder, oidcAuthenticationConfiguration))
+        .interceptCall(
+            closeStatusCapturingServerCall,
+            metadata,
+            (call, headers) -> {
+              // then
+              assertUsername().isEqualTo("test-user");
+              assertClientId().isNull();
+
+              call.close(Status.OK, headers);
+              return null;
+            });
+  }
+
+  @Test
+  public void
+      addsUsernameToOidcContextWhenPreferUsernameClaimIsTrueWhenUsernameAndClientIdClaimsArePresent() {
```

Comment:
This test case is the same as the previous one, `#addsUsernameInPreferenceToClientIdWhenPreferUsernameClaimIsTrue`.

Reply:
I've removed the duplicate test, good spot